firmware/application/apps/ui_level.cpp

```diff
@@ -170,11 +187,26 @@ void LevelView::on_statistics_update(const ChannelStatistics& statistics) {
         last_max_rssi = rssi_graph.get_graph_max();
         freq_stats_rssi.set("RSSI: " + to_string_dec_uint(last_min_rssi) + "/" + to_string_dec_uint(last_avg_rssi) + "/" + to_string_dec_uint(last_max_rssi) + ", dt: " + to_string_dec_uint(rssi_graph.get_graph_delta()));
     }
+
+    if (beep) {
+        uint32_t beep_freq = 400 + ((80 + statistics.max_db) * 7600) / 85;
+        baseband::request_audio_beep(beep_freq, 24000, 100);
```

Comment:
0.  Just FYI, there are two functions that can cause a beep, either baseband::request_rssi_beep() which requires the beep frequency calculation code in each baseband module (as in proc_sonde), or baseband::request_audio_beep() where you do the beep frequency calculation here in the ui code and the baseband code just generates the sine wave.  I think you've made a valid decision to use request_audio_beep() here, but just FYI.

1.  The middle parameter to baseband::request_audio_beep() should match whatever "audio_sampling_rate" that you've configure in audio::set_rate().  (This parameter is telling the baseband code what the audio sampling rate is set to by the ui code.)

Looks like you're using 4 different baseband modules, and I assume that you want the beep in all modes, so:

2.  In the main module of proc_capture.cpp, you'll need to add the following line (it's already in the other modules):
    
`    audio::dma::init_audio_out();
`

3.  You'll need to update the on_message handler in each baseband module that you're using to handle the Message::ID:AudioBeep message type (see proc_audio_beep.cpp for example).

Example:
```
void AudioBeepProcessor::on_message(const Message* const msg) {
    switch (msg->id) {
        case Message::ID::AudioBeep:
            on_beep_message(*reinterpret_cast<const AudioBeepMessage*>(msg));
            break;
    }
}


void AudioBeepProcessor::on_beep_message(const AudioBeepMessage& message) {
    audio::dma::beep_start(message.freq, message.sample_rate, message.duration_ms);
}
```

Reply:
Hmm looks like it's not far away. If I got it working here, I think it would be nice to add it to audio app, and maybe looking glass

Reply:
Thanks, I'll come back after a few tries

Reply:
FYI:  Regarding comment 1 above, note that the middle parameter is in hertz like you have (such as 24000), not a audio::Rate enum (such as Hz_24000).

Reply:
It works !